File: appjava/src/main/java/cn/govast/vastutilsjava/network/UserService.java
/*
 * Copyright 2022 VastGui guihy2019@gmail.com
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package cn.govast.vastutilsjava.network;

// Author: Vast Gui
// Email: guihy2019@gmail.com
// Date: 2022/9/26
// Description: 
// Documentation:
// Reference:


import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.POST;
import retrofit2.http.Query;


public interface UserService {

    @POST("/login/qr/key")
    Observable<QRCodeKey> generateQRCode(@Query("timestamp") String timestamp);

}

File: app/src/main/kotlin/cn/govast/vastutils/network/service/UserService.kt
package cn.govast.vastutils.network.service

import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

// Author: Vast Gui
// Email: guihy2019@gmail.com
// Date: 2022/9/24
// Description: 
// Documentation:

interface UserService {

    @POST("/login/qr/key")
    suspend fun generateQRCode(@Query("timestamp") timestamp:String): QRCodeKey

    @GET("/search")
    suspend fun searchSong(@Query("keywords") keywords:String,@Query("timestamp") timestamp:String): SongResult

}